pkg/network/network.go

```diff
@@ -93,47 +93,52 @@ func IsChangeSafe(prev, next *opv1alpha1.NetworkAddonsConfigSpec) error {
 	return nil
 }
 
-func Render(conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir string, openshiftNetworkConfig *osv1.Network, clusterInfo *ClusterInfo) ([]*unstructured.Unstructured, error) {
+func Render(prev, conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir string, openshiftNetworkConfig *osv1.Network, clusterInfo *ClusterInfo) ([]*unstructured.Unstructured, []*unstructured.Unstructured, error) {
```

Comment:
There is no reason why `Render` should handle anything related to removals. All it should do is to render templates from `data/` based on environment variables and data from `NetworkAddonsConfig`.

Please introduce a new function `ObejctsToRemove` which would calculate the removal.

Reply:
done